src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1341,5 +1346,51 @@ private Task<bool> InternalReceiveResponseHeadersAsync(WinHttpRequestState state
 
             return state.TcsReceiveResponseHeaders.Task;
         }
+
+        private static Guid LogRequest(HttpRequestMessage request)
+        {
+            if (_diagnosticSource.IsEnabled("System.Net.Http.Request"))
+            {
+                Guid loggingRequestId = Guid.NewGuid();
+                long timestamp = Stopwatch.GetTimestamp();
+
+                _diagnosticSource.Write(
+                    "System.Net.Http.Request",
+                    new
+                    {
+                        Request = request,
```

Comment:
Is exposing  request and response objects like this really a good thing to do? It means any code in the process can hijack these objects, manipulate the streams, etc., right? That seems dangerous.